cpp/src/io/orc/writer_impl.cu

```diff
@@ -1067,6 +1067,10 @@ void writer::impl::write(table_view const &table)
 
   auto streams  = create_streams(orc_columns, stripe_bounds);
   auto enc_data = encode_columns(*device_columns, orc_columns, str_col_ids, stripe_bounds, streams);
```

Comment:
Maybe it would be cleaner to move `dict_index` and `dict_data` into `encode_columns`. This way it's impossible to free the memory at the wrong time and it's clear that `encode_columns` consumes/takes ownership of these objects (which makes it clearer why we are releasing the memory).
The current fix is more imperative than declarative and might be hard to maintain as we refactor this function.

Reply:
forgot the reference: https://www.modernescpp.com/index.php/c-core-guidelines-how-to-pass-function-parameters